staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -282,6 +289,9 @@ func (h *UpgradeAwareHandler) tryUpgrade(w http.ResponseWriter, req *http.Reques
 		backendConn, rawResponse, err = utilnet.ConnectWithRedirects(req.Method, &location, clone.Header, req.Body, utilnet.DialerFunc(h.DialForUpgrade), h.RequireSameHostRedirects)
 	} else {
 		klog.V(6).Infof("Connecting to backend proxy (direct dial) %s\n  Headers: %v", &location, clone.Header)
+		if h.OverrideRequestHost {
+			clone.Host = h.Location.Host
```

Comment:
does code coverage show this line being exercised in unit tests?

Reply:
@liggitt it was not. I added a new test case called `use location host - invalid upgrade` to `TestServeHTTP`, that covers it and tests this explicitly.